### PR TITLE
Refactor signup stage notifications job and add tests

### PR DESCRIPTION
**Pull request by craigspaeth**
Apologies for going a little overboard on refactoring your code here, but hopefully this is a useful demonstration of my approach to refactoring and test coverage. I'll comment more inline.

### CODE REVIEW

File: apps/jobs/test/signup-stage-notifications.test.js [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+})
+
+test('notificationName gets the notification name for lead', () => {
+  const lead = { id: 'karen', signupStage: 'schedulingCoaching' }
+  expect(notificationName(lead)).toEqual('scheduleSession')
+  const lead2 = { id: 'karen', signupStage: 'referral' }
+  expect(notificationName(lead2)).toEqual('referralWelcome')
+})
+
+test('main sends notifications email, sms, for signup stage', async () => {
```
Comment (craigspaeth) [inline apps/jobs/test/signup-stage-notifications.test.js]: This test would be considered an integration test as it's testing a long complicated procedure combining our "pure business logic" with IO calls to services.

As you can see there's a lot of mocking ceremony and difficulty that comes with writing these kinds of tests higher up the stack so I'm keeping coverage minimal to this one deep test against the `default` function and letting the unit tests provide wide and shallow coverage. I encourage this ["practical testing pyramid"](https://martinfowler.com/articles/practical-test-pyramid.html) philosophy as a way to keep tests running fast, predictable, and with sufficient coverage. I've seen too many times how not writing enough tests or too many integration/end-to-end tests can destroy a project over time. So I find striking the right balance is really important to a team's velocity—a shallow pyramid is the way to go IMO: an abundance of unit tests, some integration tests, very few end-to-end tests.

File: lib/services/index.js [hunk 2]
```diff
@@ -4,16 +4,26 @@
...
+  MAILGUN_DOMAIN,
+  MAILGUN_KEY,
+  STRIPE_SECRET,
+  TWILIO_ACCOUNT_SID,
+  TWILIO_TOKEN
 } = process.env
 
+export const twilio = _twilio(TWILIO_ACCOUNT_SID, TWILIO_TOKEN)
+
+export const mailgun = _mailgun({ apiKey: MAILGUN_KEY, domain: MAILGUN_DOMAIN })
```
Comment (craigspaeth) [inline lib/services/index.js]: Moving twilio and mailgun to this little services wrapper lib DRYs up the boilerplate and makes it a lot easier to mock in tests.

**Comment by lionsdwarf** [comment]
Understood--thanks for the heads up. I will be conscientious to follow this pattern going forward.

**Comment by craigspaeth** [comment]
@lionsdwarf Sounds good! To be clear I'm not suggesting that all code has to be written to this degree of being functional/buttoned-up/test-covered. I just wanted to add tests back to this job and got a bit carried away refactoring. I hope it's useful peek at my approach and not taken as undermining any prior code or other philosophies/styles—the code you had in there was excellent and did all the hard work of getting me this far.

**Comment by lionsdwarf** [comment]
lol. Totally understand--no worries, we are on the same page @craigspaeth!

**Pull request merged**